lib/checkPullRequestTemplate.js

```diff
@@ -165,69 +170,144 @@ const checkEssentialChecklist = (pullRequest) => {
           pullRequest.head.repo.full_name === pullRequest.base.repo.full_name;
         if (!pullRequest.maintainer_can_modify && !isPrFromFork) {
-          if (message) {
-            message +=
-              'The allow edits from maintainers checklist needs to ' +
-              'be ticked so that maintainers can rerun failed tests. ' +
-              'Endeavour to add this by ticking on the check box. ';
-          } else {
-            message +=
-              'the allow edits from maintainers checklist needs to ' +
-              'be ticked so that maintainers can rerun failed tests. ' +
-              'Endeavour to add this by ticking on the check box. ';
-          }
+          message +=
+            'The allow edits from maintainers checklist needs to ' +
+            'be ticked so that maintainers can rerun failed tests. ' +
+            'Endeavour to add this by ticking on the check box. ';
         }
       }
     });
 
     return message;
   } else {
     return (
-      'the body of this PR is missing the checklist section, please ' +
+      'The body of this PR is missing the checklist section, please ' +
       'update it to include the checklist.'
     );
   }
 };
 
+
+/**
+ * This function checks if the pull request's body contains the
+ * proof of changes and returns an appropriate message if the required
+ * proofs are not provided.
+ *
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const checkProofThatChangesAreCorrect = (pullRequest) => {
+  /*
+    A valid Proof that changes are correct section would look like:
+    ## Proof that changes are correct
+
+    <!--
+    Add videos/screenshots of the user-facing interface to demonstrate
+    that the changes made in this PR work correctly.
+    -->
+    ![oppiabot-test](https://user-images.githubusercontent.com/30050862/
+      test.gif)
+    (or)
+    https://user-images.githubusercontent.com/30050862/
+      test.mp4
+  */
+  const proofOfChangesHeadingIndex = pullRequest.body.indexOf(PROOF_TEXT);
+
+  if (proofOfChangesHeadingIndex !== -1) {
+    const nextHeadingIndex = pullRequest.body.indexOf(
+      HEADING_START,
+      proofOfChangesHeadingIndex + PROOF_TEXT.length
+    );
+
+    let proofOfChangesSection = pullRequest.body.substring(
+      proofOfChangesHeadingIndex,
+      nextHeadingIndex
+    );
+
+    const proofIsNotPossible = proofOfChangesSection.replace(
+      PROOF_HTML_COMMENT, ''
+    ).match(PROOF_UNAVAILABLE_TEXT);
+    console.log(proofOfChangesSection.replace(
+      PROOF_HTML_COMMENT, ''
+    ).match(PROOF_UNAVAILABLE_TEXT));
+    const proofLinks = proofOfChangesSection.includes(UPLOADED_MEDIA);
+
+    // Check if there are images/videos added in the section.
+    if (!proofLinks && !proofIsNotPossible){
+      return (
+        'The proof that changes are correct has not been provided, ' +
+        'please make sure to upload a image/video showing that the changes ' +
+        'are correct.'
```

Comment:
```suggestion
        'The proof that changes are correct has not been provided, ' +
        'please make sure to upload a image/video showing that the changes ' +
        'are correct. Or include a sentence saying "No proof of changes' +
        'needed because" and the reason why proof of changes cannot be provided.'
```

Reply:
DONE